pkg/controller/base_controller.go

```diff
@@ -159,8 +159,12 @@ func (c *controller) processNextWorkItem() bool {
 		c.workQueue.Forget(key)
 	}
 
-	err := c.reconcile(key)
+	shouldRequeue, err := c.reconcile(key)
 	if err == nil {
+		if shouldRequeue {
+			c.workQueue.Add(obj)
```

Comment:
should this add 'key' or 'obj'?
In enqueue, 'key' got added to workqueue.
(or) instead of calling 'Add', call 'enqueue' with 'obj'

Reply:
This will be `obj`. We are re-adding the same `obj`, we fetched from workqueue. We can't call `enqueue` here since this function is not aware of the custom resource object.

Reply:
obj and key are same